Added new whereDoesntHaveAny method

AlexPHPower opened the pull request:
# What?

This is just a simple addition to the doesntHave method.

# How?

By adding a `whereDoesntHaveAny()` function that loops over the passed array of relations and returns all records that do not have the provided relations.

# Why?

If you wanted to retrieve records but exclude those with a set of relations you'd previously have to chain multiple `whereDoesntHave()` queries in order to remove all records with your undesired relations.

# Who?

I'm quite new to web development and have been a professional developer for 4 years and this is my first open source contribution.

I apologise in advance for anything i may have gotten wrong.

Commit:
Added new whereDoesntHaveAny method

dennisprudlo commented on src/Illuminate/Database/Eloquent/Concerns/QueriesRelationships.php:
```diff
@@ -348,6 +348,18 @@ public function orWhereDoesntHaveMorph($relation, $types, Closure $callback = nu
+    public function whereDoesntHaveAny($relations, Closure $callback = null)
+    {
+        foreach ($relations as $relation) {
+            return $this->doesntHave($relation);
```
This will only be executed for the first relation of the array. Also the `$callback` argument is never used.

AlexPHPower replied:
Yeah sorry i created the pr as a draft cause i'm not sure how to run the tests locally otherwise i'd have done this locally before creating 🙈.

Thanks for the pointers!

Commit:
Made changes to the test

Realised my error in how logically i was testing the query

Commit:
Added a closure so that the function loops

Commit:
Added images function to comments

Commit:
Fixed orWhereRelation based on new post created

Commit:
changed comment on closure

This part is repeated code that i've used from hasNested so changed the comment to reflect this

taylorotwell commented:
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.
